resources/assets/js/components/ChannelDropdown.vue

```diff
@@ -0,0 +1,34 @@
+<template>
+    <li class="dropdown">
+        <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true"
+           aria-expanded="false">Channels <span class="caret"></span></a>
+
+        <div class="dropdown-menu" >
+            <div style="padding:.5rem 1rem">
+                <input type="text" class="form-control" v-model="filter" placeholder="Filter Channels..."/>
+            </div>
+            <ul style="max-height: 400px; overflow:auto; ">
+                <li v-for="channel in filteredThreads"><a :href="'/threads/' + channel.slug" v-text="channel.name"></a>
+                </li>
+            </ul>
+        </div>
+    </li>
+</template>
+
+<script>
+    export default {
+        props: ['channels'],
+        data() {
```

Comment:
To stay consistent with other components I would add a new line after the `props` array and `},` of `data()`